modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryModelListener.java

```diff
@@ -64,6 +64,22 @@ private void _validate(ObjectEntry objectEntry) {
 					"an-api-sort-must-be-related-to-an-api-endpoint");
 			}
 
+			if (Validator.isNull(
+					_objectEntryHelper.getObjectEntry(
```

Comment:
This can be simplified just fetching by ID and checking the retrieveType to save computation. Same in the other class.

Reply:
Done!